spec/services/data_migrations/migrate_to_structured_offer_conditions_spec.rb

```diff
@@ -0,0 +1,35 @@
+require 'rails_helper'
+
+RSpec.describe DataMigrations::MigrateToStructuredOfferConditions do
+  around do |example|
+    Timecop.freeze do
+      example.run
+    end
+  end
+
+  before do
+    @reference_condition = create(:reference_condition, updated_at: 1.day.ago)
+    @ske_condition = create(:ske_condition, updated_at: 1.day.ago)
+    @text_condition = create(:text_condition, updated_at: 1.day.ago)
+    @offer_condition = create(:text_condition, updated_at: 1.day.ago)
+    @offer_condition.update_columns(type: nil, text: 'Learn to dance', details: nil)
+  end
+
+  it 'does not alter existing structured condition records' do
+    described_class.new.change
+
+    [@reference_condition, @ske_condition, @text_condition].each do |structured_condition|
+      expect(structured_condition.reload.updated_at).to be_within(1.second).of(1.day.ago)
```

Comment:
given you're freezing time in the `before` block, is `be_within` necessary here?

Reply:
as we talked about earlier, the `be_within` seems to be necessary due to the difference in precision between PostgreSQL and Ruby timestamps.